pkg/messaging/message.proto

```diff
@@ -14,4 +14,5 @@ message Message {
 	string protocol  = 4;
 	bytes  payload   = 5;
 	int64  created   = 6; // Unix timestamp in nanoseconds
+	bytes  span      = 7;
```

Comment:
It is a bit harder to pass the span context on the marshalled message as on the subscriber side the span context ended up being corrupted. That is why I added the additional field. We can try and do without but it is not possible to propagate the span context on an outbound message without a carrier. 
a possible less invasive way would be to wrap the message and span context in a map[string]interface{} over the broker. Something like:
```
"data":[]byte //the Message
"span":[]byte // span context
```

Reply:
@SammyOina The other approach would be to modify interfaces to accept context. Tracing is not necessarily related to the message, but to its metadata. It could, eventually, be a part of the message, but what it really is is - message context, so what we can do as an alternative is to pass context to our interfaces. It is as intrusive as it gets, but at least it conceptually decouples tracing from the message and provides a clean interface to implement.

Reply:
That was the original approach I had used and I can switch to that